infoschema/tables_test.go

```diff
@@ -769,3 +802,18 @@ func (s *testTableSuite) TestForTableTiFlashReplica(c *C) {
 	tbl.Meta().TiFlashReplica.Available = true
 	tk.MustQuery("select TABLE_SCHEMA,TABLE_NAME,REPLICA_COUNT,LOCATION_LABELS,AVAILABLE from information_schema.tiflash_replica").Check(testkit.Rows("test t 2 a,b 1"))
 }
+
+func (s *testClusterTableSuite) TestForClusterServerInfo(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	mockAddr := "127.0.0.1:10080"
+	instances := []string{
+		"pd,127.0.0.1:11080," + mockAddr,
+		"tidb,127.0.0.1:11080," + mockAddr,
+		"tikv,127.0.0.1:11080," + mockAddr,
+	}
+	fpExpr := `return("` + strings.Join(instances, ";") + `")`
+	c.Assert(failpoint.Enable("github.com/pingcap/tidb/infoschema/mockClusterInfo", fpExpr), IsNil)
```

Comment:
I have some concerns about this failpoint, which may cause unit test unstable because the deferent test case shares the same failpoint. It's better to refactor the failpoint logic and use `failpoint.InjectContext` to avoid interleaving between deferent test suites.

Reply:
Already use a `SerialSuites` to avoid this.